lib/response.js

```diff
@@ -717,9 +717,14 @@ res.header = function header(field, val) {
       : String(val);
 
     // add charset to content-type
-    if (field.toLowerCase() === 'content-type' && !charsetRegExp.test(value)) {
-      var charset = mime.charsets.lookup(value.split(';')[0]);
-      if (charset) value += '; charset=' + charset.toLowerCase();
+    if (field.toLowerCase() === 'content-type') {
+      if (Array.isArray(value)) {
```

Comment:
The value is guaranteed to be a string or an array of strings: https://github.com/oztune/express/blob/80a720753589ab4ecb12a266f137493f794e94c9/lib/response.js#L717